tests/lib/linter.js

```diff
@@ -1664,7 +1664,7 @@ describe("Linter", () => {
             assert.equal(messages[1].column, 19);
         });
 
-        it("should not report a violation", () => {
+        it("should report a violation", () => {
```

Comment:
These unit test changes makes me little uncomfortable.... Ideally refactor should not change unit test...
I am questioning to better understand the intent...

Reply:
I agree that in general, refactors shouldn't change unit tests.

However, I think these tests were asserting incorrect behavior. The first test was asserting that no error should be reported in cases like this:

```js
/* eslint no-alert: error */
/* eslint-disable */
/* eslint-disable */

/* eslint-enable */

alert('foo');
```

I don't think this makes sense because `eslint-enable` is supposed to enable all disabled rules, regardless of how the rules were disabled previously.

The second test was asserting that no error should be reported in cases like this:

```js
/* eslint no-alert: error */

/* eslint-disable no-alert */

/* eslint-disable no-console */

/* eslint-enable */

alert('foo');
```

I don't think this makes sense because `/* eslint-enable */` is supposed to enable all disabled rules, regardless of how many rules have been disabled before. (Previously, the user would need to use a second `/* eslint-enable */` comment to enable `no-alert` again.)